libs/service-portal/access-control/delegations/src/components/access/AccessDeleteModal.tsx

```diff
@@ -21,11 +32,25 @@ export const AccessDeleteModal = ({
   onDelete,
   ...rest
 }: AccessDeleteModalProps) => {
-  const { formatMessage } = useLocale()
+  const { formatMessage, lang } = useLocale()
   const { userInfo } = useAuth()
   const [error, setError] = useState(false)
   const [deleteAuthDelegation, { loading }] = useDeleteAuthDelegationMutation()
 
+  const {
+    data: scopeTreeData,
+    loading: scopeTreeLoading,
+  } = useAuthScopeTreeQuery({
+    variables: {
+      input: {
+        domain: delegation?.domain.name,
```

Comment:
Because the delegation is not set during rendering of the component. And if I put a conditional  around the AccessDeleteModal comp then this addon you added closes component right after I open it.
```
  const handleOnVisibilityChange = (isVisible: boolean) => {
    if (isVisible) {
      headingRef.current?.focus()
    } else {
      onClose?.()
    }
  }
```

Reply:
Ah! I remember this! That's annoying... 🙈

Still, the prop should then be conditional (`delegation?: AuthCustomDelegation`), and we need to lazily query the scope tree only if there is some `delegation.domain`.

Should also double check that `scopeTreeData` is immediately cleared while re-querying a new domain.